### ISSUE TITLE
Improve and Experiment

### ISSUE THREAD

**Issue by Nishant001-jyo** [Nishant001-jyo/Nishant001-jyo#9]
`from sklearn.model_selection import cross_val_score
scores = cross_val_score(model, X, y, cv=5)
print("Cross-Validation Accuracy:", scores.mean())`

Try other models: Import and swap (e.g., from sklearn.tree import DecisionTreeClassifier).
Visualize decision boundaries (advanced: Use matplotlib to plot 2D features).
Handle edge cases: What if input features are invalid (e.g., negative values)? Add checks.
